lm_eval/api/task.py

```diff
@@ -1642,20 +1648,22 @@ def construct_requests(self, doc: dict, ctx: Optional[str], **kwargs):
         )
 
     def process_results(self, doc: dict, results: Tuple[float]) -> dict:
-        (loglikelihood,) = results
+        (loglikelihood, tokens_) = results[0]
```

Comment:
why the [0] here?

Reply:
The reason to use `results[0]` is not straightforward, but is based on the change performed in [`hugggingface.py`](https://github.com/dtamayo-nlp/lm-evaluation-harness/blob/main/lm_eval/models/huggingface.py#L908) where instead of only saving the loglikelihood I also saved the number of tokens used in a list of N tuples [(loglikelihood_1, n_tokens_1),...,(loglikelihood_N, n_tokens_N)], where N is the number of samples. That creates a difference in the way the class Instance is saved. In each instance the attributes `resps` and `filtered_resps` will have the following formats: `resps=[(-101136.0, 42378)], filtered_resps={'none': (-101136.0, 42378)}` instead of the previous format: `resps=[-101136.0], filtered_resps={'none': -101136.0}`.

Then, when using the function [process_results](https://github.com/dtamayo-nlp/lm-evaluation-harness/blob/main/lm_eval/evaluator.py#L502), if we are using `[req.filtered_resps[filter_key] for req in requests]` we must call requests[0] to output the format (loglikelihood, _tokens) instead of [(loglikelihood, _tokens)]. 

This is the less intrusive way that I saw to implement this. However, as I see, there will be problems related to the unit tests given that you are comparing to the original format [here](https://github.com/EleutherAI/lm-evaluation-harness/blob/main/tests/models/test_huggingface.py#L129).